internal/controller/rbac/provider/roles/roles.go

```diff
@@ -154,14 +155,22 @@ func RenderClusterRoles(pr *v1.ProviderRevision, rs []Resource) []rbacv1.Cluster
 		Rules: withVerbs(rules, verbsEdit),
 	}
 
+	viewRules := rules
+	if pkgmetav1.CapabilitiesContainFuzzyMatch(pr.GetCapabilities(), pkgmetav1.ProviderCapabilitySafeStart) {
+		viewRules = append(viewRules, rbacv1.PolicyRule{
+			APIGroups: []string{"apiextensions.k8s.io"},
+			Resources: []string{"customresourcedefinitions"},
+		})
+	}
+
 	view := &rbacv1.ClusterRole{
 		ObjectMeta: metav1.ObjectMeta{
 			Name: namePrefix + pr.GetName() + nameSuffixView,
 			Labels: map[string]string{
 				keyAggregateToView: valTrue,
 			},
 		},
-		Rules: withVerbs(rules, verbsView),
+		Rules: withVerbs(viewRules, verbsView),
```

Comment:
ah, another of my comments got left off - i think because i've been adding comments to each commit in separate tabs here - not to self, don't do that anymore 😦 

so can you help me understand why this `Rules` field didn't need to be set before? I would have thought it would have previously been something like `Rules: withVerbs(rules, verbsView)`, but it wasn't being set at all. Why is that?

Reply:
I am adding a resource to viewRules above to be able to view CRDs but I don't want the provider to be given RBAC to manage CRDs.

Reply:
oh jeez, nope this comment was because i'm an idiot - i've been using the "split" view in PRs recently and i'm still getting used to it sometimes. My eyes only saw this as an Added line instead of a changed line - so I incorrectly thought `Rules:` wasn't set at all before and I was very confused. Yes, this makes total sense to update the line as you have done, sorry for the confusion! 😊